test/test_compiler_opts.py

```diff
@@ -72,6 +72,29 @@ def test_opts_stanc(self):
             ['STANCFLAGS+=--warn-uninitialized', 'STANCFLAGS+=--name=foo'],
         )
 
+        stanc_opts['O1'] = True
+        opts = CompilerOptions(stanc_options=stanc_opts)
+        opts.validate()
+        self.assertEqual(
+            opts.compose(),
+            ['STANCFLAGS+=--warn-uninitialized',
+             'STANCFLAGS+=--name=foo',
+             'STANCFLAGS+=--O1'],
+             )
+
+        # should add to logger
```

Comment:
We sometimes use LogCapture in the tests to make sure that warnings are emitted. Here's an example:
https://github.com/stan-dev/cmdstanpy/blob/2b14038979da9b6490d215221575b3592e7dae29/test/test_cmdstan_args.py#L580-L596

Reply:
everything passes locally
```base
test/test_compiler_opts.py::CompilerOptsTest::test_opts_add PASSED                       [  8%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_add_include_paths PASSED         [ 16%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_cpp PASSED                       [ 25%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_cpp_opencl PASSED                [ 33%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_empty PASSED                     [ 41%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_empty_eq PASSED                  [ 50%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_stanc PASSED                     [ 58%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_stanc_deprecated PASSED          [ 66%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_stanc_ignore PASSED              [ 75%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_stanc_includes PASSED            [ 83%]
test/test_compiler_opts.py::CompilerOptsTest::test_opts_stanc_opencl PASSED              [ 91%]
test/test_compiler_opts.py::CompilerOptsTest::test_user_header PASSED                    [100%]


```